### PR TITLE
Optimize navigations in settings in some special cases

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-2248

- Added a query `q_from` to store the originating url, so that the back button can back to there.

### CODE REVIEW

File: pkg/auth/webapp/result.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ func (r *Result) WriteResponse(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// Propagate q_from
+	if req.URL.Query().Get(QueryFrom) != "" {
+		q.Set(QueryFrom, req.URL.Query().Get(QueryFrom))
+	}
```
Comment (louischan-oursky) [inline pkg/auth/webapp/result.go:51]: Wouldn't this make `q_from` like `x_from`?

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/auth/webapp/result.go (hunk 1 above)
Comment (tung2744) [reply pkg/auth/webapp/result.go:51]: We only propagate q_from in an redirect response, but not in any page.
Rationale: When user POST something to an endpoint, and that endpoint returned an redirect response. In the user's perspective the user is reaching the redirected uri from the page he POST the request.

Comment (tung2744) [reply pkg/auth/webapp/result.go:53]: Added a comment to explain it

**Comment by louischan-oursky** [comment]
Looks good to me. Please fix the build.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**